common/src/Polyhedron_ConvexHull.h

```diff
@@ -418,16 +418,20 @@ typename Polyhedron<T,FP,VP>::Vertex* Polyhedron<T,FP,VP>::addPointToPolygon(con
     HalfEdge* firstEdge = face->boundary().front();
     HalfEdge* curEdge = firstEdge;
     do {
-        HalfEdge* previous = curEdge->previous();
-        HalfEdge* next = curEdge->next();
-        if ((previous->pointStatus(facePlane.normal, position) == Math::PointStatus::PSBelow &&
-              curEdge->pointStatus(facePlane.normal, position) != Math::PointStatus::PSBelow)) {
+        HalfEdge* prevEdge = curEdge->previous();
+        HalfEdge* nextEdge = curEdge->next();
+        const Math::PointStatus::Type prevStatus = prevEdge->pointStatus(facePlane.normal, position);
+        const Math::PointStatus::Type  curStatus =  curEdge->pointStatus(facePlane.normal, position);
+        const Math::PointStatus::Type nextStatus = nextEdge->pointStatus(facePlane.normal, position);
+        
+        const bool curContains = curStatus == Math::PointStatus::PSInside && position.containedWithinSegment(curEdge->origin()->position(), curEdge->destination()->position());
+        
+        if (prevStatus == Math::PointStatus::PSBelow &&  curStatus != Math::PointStatus::PSBelow && !curContains)
```

Comment:
Yes to the former, but no to the latter: The point can still be on the line through both points of the current edge, but not within the segment. If you made that change, `PolyhedronTest::testAddPointToRectangleMakingOneColinear` fails.